src/arch/LowLevelWindow/LowLevelWindow_SDL.cpp

```diff
@@ -0,0 +1,258 @@
+    }
+#endif
+
+    SDL_Init(SDL_INIT_VIDEO | SDL_INIT_EVENTS | SDL_INIT_JOYSTICK | SDL_INIT_GAMEPAD);
+}
+
+LowLevelWindow_SDL::~LowLevelWindow_SDL()
+{
+    SDL_GL_DestroyContext(m_pGlContext);
+    SDL_DestroyWindow(m_pWindow);
+    SDL_Quit();
+}
+
+void* LowLevelWindow_SDL::GetProcAddress(RString s)
+{
+    return SDL_GL_GetProcAddress(s);
+}
+
+RString LowLevelWindow_SDL::TryVideoMode(const VideoModeParams& p, bool& bNewDeviceOut)
+{
+    // Ensure we're using compat profile
+    SDL_GL_SetAttribute(SDL_GL_CONTEXT_PROFILE_MASK, SDL_GL_CONTEXT_PROFILE_COMPATIBILITY);
+
+    if (!m_pWindow)
+    {
+        m_pWindow = SDL_CreateWindow(p.sWindowTitle, p.width, p.height, SDL_WINDOW_OPENGL | SDL_WINDOW_HIDDEN | SDL_WINDOW_HIGH_PIXEL_DENSITY);
+        if (!m_pWindow)
+        {
+            LOG->Info("[LLW_SDL::TryVideoMode] Failed to create SDL Window: %s", SDL_GetError());
+            return "Failed to create SDL Window - Check logs";
+        }
+
+        SDL_SetWindowPosition(m_pWindow, SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED);
+        m_pGlContext = SDL_GL_CreateContext(m_pWindow);
+
+        if (!m_pGlContext)
+        {
+            LOG->Info("[LLW_SDL::TryVideoMode] Failed to create GL Context: %s", SDL_GetError());
+            return "Failed to create GL Context - Check logs";
+        }
+
+        SDL_GL_MakeCurrent(m_pWindow, m_pGlContext);
+        SDL_ShowWindow(m_pWindow);
+
+        // New window created
+        bNewDeviceOut = true;
+    } else
+    {
+        bNewDeviceOut = false;
+    }
+
+    // Adjust for any hidpi scaling & prevent division by zero
+    // Note: This is rounded to the nearest 4 decimals to prevent improperly set scale values
+    auto windowScale = std::floor(SDL_GetWindowDisplayScale(m_pWindow) * 1000) / 1000;
+    if (windowScale <= 0.01f)
+    {
+        windowScale = 1.0f;
+    }
+    SDL_SetWindowSize(m_pWindow, static_cast<int>(static_cast<float>(p.width) / windowScale), static_cast<int>(static_cast<float>(p.height) / windowScale));
+
+    // Retrieve the real pixel size in case we have a high-dpi display
+    int widthInPixels = p.width;
+    int heightInPixels = p.height;
+
+    // We want the currentDisplay (monitor), the current display mode (resolution),
+    // and the real window size in pixels (ignoring hidpi)
+    auto currentDisplay = SDL_GetDisplayForWindow(m_pWindow);
+    auto currentDisplayMode = SDL_GetCurrentDisplayMode(currentDisplay);
+    SDL_GetWindowSizeInPixels(m_pWindow, &widthInPixels, &heightInPixels);
+
+    LOG->Info("[LLW_SDL::TryVideoMode] Adjusting for HiDPI scale of %f. Requested: [%dx%d] Real: [%dx%d]", windowScale,
+              p.width, p.height, widthInPixels, heightInPixels);
+
+    // Adjust VSync, fullscreen, and borderless modes if needed
+    SDL_GL_SetSwapInterval(p.vsync);
+    SDL_SetWindowFullscreen(m_pWindow, !p.windowed);
+    SDL_SyncWindow(m_pWindow);
+    SDL_SetWindowBordered(m_pWindow, !p.windowed || !p.bWindowIsFullscreenBorderless);
```

Comment:
I didn't debug these, but there are couple issues:

1. when using native resolution with borderless the window is not positioned in (0, 0), but slightly off:
![image](https://github.com/user-attachments/assets/c067104f-8b61-4835-9c6c-7c9c9decaa63)


2. when using fullscreen with a resolution smaller than native, the game still uses native one, but renders only a corner:

![image](https://github.com/user-attachments/assets/9d11992a-a924-48e7-be0d-c1a5748438ed)

3. there seem to be no difference between fullscreen and borderless

4. windowed in smaller than native res doesn't seem to scale contents to the window, but full restart of the game seems to help

![image](https://github.com/user-attachments/assets/cde607e8-6d0d-44c4-8b58-3c96c3dbc215)